frontend/public/components/utils/promise-component.tsx

```diff
@@ -4,19 +4,20 @@ export const withHandlePromise: WithHandlePromise = (Component) => (props) => {
   const [inProgress, setInProgress] = React.useState(false);
   const [errorMessage, setErrorMessage] = React.useState('');
 
-  const handlePromise = (promise) => {
+  const handlePromise = (promise, onFulfill, onError) => {
     setInProgress(true);
-    return promise.then(
+    promise.then(
       (res) => {
         setInProgress(false);
         setErrorMessage('');
-        return res;
+        onFulfill && onFulfill(res);
       },
       (error) => {
         const errorMsg = error.message || 'An error occurred. Please try again.';
         setInProgress(false);
         setErrorMessage(errorMsg);
-        return Promise.reject(errorMsg);
+        // eslint-disable-next-line no-console
+        onError ? onError(error) : console.error(error);
```

Comment:
`onError` callback should be backwards compatible. I just mean changing the console.error